lib/upload.js

```diff
@@ -85,20 +85,32 @@ class Upload {
       options = {};
     }
 
+    let retryAttempt = 0;
+    const retryIfShould = (err) => {
+      if (shouldRetry(err, retryAttempt, options)) {
+        const delay = options.retryDelays[retryAttempt++];
+        setTimeout(() => Upload.terminate(url, options, cb), delay);
+        return;
+      } else {
+        return cb(err);
```

Comment:
This return statement can be removed as well.

Reply:
resolved